src/engine/InputQuery.cpp

```diff
@@ -157,6 +163,54 @@ void InputQuery::addPiecewiseLinearConstraint( PiecewiseLinearConstraint *constr
     _plConstraints.append( constraint );
 }
 
+void InputQuery::addClipConstraint( unsigned b, unsigned f,
```

Comment:
Do we need to support infinite upper and lower bounds?  like floor=-infinity or ceiling=+infinity？